runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -244,6 +271,24 @@ private boolean findSharedCore() {
         return true;
     }
 
+    private boolean findDownloadedCore() {
+        String libDir = mWrapperContext.getDir(PRIVATE_LIB_DIR, Context.MODE_PRIVATE).toString();
+        String dexPath = libDir + File.separator + "classes.dex";
+        String dexOutputPath = mWrapperContext.getDir("dex", 0).getAbsolutePath();
```

Comment:
Per the API doc:
Use 0 or MODE_PRIVATE for the default operation, MODE_WORLD_READABLE and MODE_WORLD_WRITEABLE to control permissions.

Yet, it would be good to use MODE_PRIVATE explicitly.